Update the usage of chainID in lip0066 (#228)
LiskHQ/lips: proposals/lip-0066.md

| `PrivateKeyEd25519` | bytes  | Byte sequences of length `ED25519_PRIVATE_KEY_LENGTH`. | An Ed25519 private key. |
| `PrivateKeyBLS`     | bytes  | Byte sequences of length `BLS_PRIVATE_KEY_LENGTH`. | An BLS private key. |

## uint32be Function
We define the function `uint32be` to convert bytes to unsigned integer values.

```python

> def uint32be(chainID: bytes) -> int:

Reviewer: minor comment, but in other LIPs we use the type `uint32`, so we could use that in the output type instead of `int`

Author: Updated.

> def uint32be(chainID: bytes) -> int:

Reviewer: I just realize I made a mistake with my previous comment: the function `uint32be` we define in other LIPs converts from uint32 to bytes, here instead we want to convert from bytes to uint32, hence the previous name you had was fine, sorry about that.

Author: Replaced with the old one.

| `PrivateKeyEd25519` | bytes  | Byte sequences of length `ED25519_PRIVATE_KEY_LENGTH`. | An Ed25519 private key. |
| `PrivateKeyBLS`     | bytes  | Byte sequences of length `BLS_PRIVATE_KEY_LENGTH`. | An BLS private key. |

> ## bytesToInt Conversion

Reviewer: ```suggestion
## uint32be
```

Reviewer: In the past we used this function name to convert bytes to uint

Author: Replaced.

> For the `chainID` , we use the bytes value. We define the function `bytesToInt` to deserialize bytes to integer value.

Reviewer: ```suggestion
We define the function `uint32be` to convert bytes to unsigned integer values.
```

Author: Done.

```python

> def bytesToInt(numberBytes: bytes) -> int:

Reviewer: we want to convert chainID to an integer, why isnt the input just `chainID`?

Author: Updated the function. Although, input could be anything.

>     if len(numberBytes) > chainID:

Reviewer: What is chainID here?

Author: updated the function.

>         raise Exception()

Reviewer: Exceptions should have a descriptive error message

Author: Updated the function.

>     return big-endian decoding of numberBytes

Reviewer: Let's use python code here `int.from_bytes( chainID, "big")` as we already do in other LIPs

Author: Updated.

The path for deriving the n-th private key from that phrase is `m/44'/134'/n'`. This follows the specifications of [BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Registered_coin_types) and [SLIP 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md), assigning 134 to the LSK token.

> The path for deriving the private key of the n-th [generator key](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#generator-key) pair, for the chain with [ID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0043.md#chain-id) `ChainID` is `m/25519'/134'/ChainIDConverted'/n'`, where `ChainIDConverted` is the return integer value of `uint32be(chainID)`. The hardened derivation `chainIDConverted'` can always be converted to a byte value that fits into 4 bytes because the first bit is set to `0`.

Reviewer: ```suggestion
The path for deriving the private key of the n-th [generator key](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#generator-key) pair, for the chain with [ID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0043.md#chain-id) `chainID` is `m/25519'/134'/chainIDConverted'/n'`, where `chainIDConverted` is the return integer value of `uint32be(chainID)`. The hardened derivation `chainIDConverted'` can always be converted to a byte value that fits into 4 bytes because the first bit is set to `0`.
```

Author: Updated the text.

> The path for deriving the private key of the n-th [generator key](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#generator-key) pair, for the chain with [ID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0043.md#chain-id) `ChainID` is `m/25519'/134'/ChainIDConverted'/n'`, where `ChainIDConverted = bytesToUint(chainID)`. The hardened value `chainIDConverted'` will always be within the range of a `uint32` because [the first bit of a chainID must be 0](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#chain-identifiers-1).

Reviewer: ```diff 
The path for deriving the private key of the n-th [generator key](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#generator-key) pair, for the chain with [ID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0043.md#chain-id) `ChainID` is `m/25519'/134'/ChainIDConverted'/n'`, where `ChainIDConverted = bytesToUint(chainID)`. The hardened value `chainIDConverted'` will always be within the range of a `uint32` because [the first bit of a chain ID must be 0](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#chain-identifiers-1).
```

The path for deriving the n-th private key from that phrase is `m/44'/134'/n'`. This follows the specifications of [BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Registered_coin_types) and [SLIP 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md), assigning 134 to the LSK token.

> The path for deriving the private key of the n-th [generator key](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#generator-key) pair, for the chain with [ID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0043.md#chain-id) `chainID `is `m/25519'/134'/chainID'/n'.` Here `chainID` is the big-endian decoding of `bytesToInt(chainID)`.

Reviewer: I would change the variable name, otherwise you are calling chainID both the bytes value and the uint representation

Author: Changed the variable name 'chainID' to 'chainIDConverted'

#### Derivation Path for BLS Keys

> Similarly to the specifications of [EIP 2334](https://eips.ethereum.org/EIPS/eip-2334), we use the path `m/12381/134/chainIDConverted/0` for the BLS key derived for the chain with ID `chainID` where `ChainIDConverted` is the return integer value of `uint32be(chainID)`.

Reviewer: ```suggestion
Similarly to the specifications of [EIP 2334](https://eips.ethereum.org/EIPS/eip-2334), we use the path `m/12381/134/chainIDConverted/0` for the BLS key derived for the chain with ID `chainID` where `chainIDConverted = uint32be(chainID)`.
```

Author: Updated the text.

#### Derivation Path for BLS Keys

> Similarly to the specifications of [EIP 2334](https://eips.ethereum.org/EIPS/eip-2334), we use the path `m/12381/134/chainID/0` for the BLS key derived for the chain with ID `chainID`. Here `chainID` is the big-endian decoding of `bytesToInt(chainID)`.

Reviewer: same as above

Author: Changed the variable name 'chainID' to 'chainIDConverted'
